dvc/command/metrics.py

```diff
@@ -12,7 +12,46 @@ def _show(self, metrics):
                 logger.info("{}:".format(branch))
 
             for fname, metric in val.items():
-                logger.info("\t{}: {}".format(fname, metric))
+                self._output_formatted_metric(fname, metric)
+
+    def _is_json(self, content):
+        import json
+
+        try:
+            json.loads(content)
+            return True
+        except ValueError:
+            return False
+
+    def _output_formatted_metric(self, fname, metric):
+        import csv
+
+        if "\n" not in metric:
+            logger.info("\t{}: {}".format(fname, metric))
+            return
+
+        logger.info("\t{}:".format(fname))
+
+        if self._is_json(metric):
+            for line in metric.split("\n"):
+                logger.info("\t\t{}".format(line))
+            return
+
+        reader = csv.reader(
+            metric.split("\n"), dialect=csv.Sniffer().sniff(metric)
```

Comment:
why split("\n") here?

Reply:
The `csv.reader` can't detect newlines in the content, actually, the first argument should be an iterator, and I think file objects uses `readline()` internally as an iterator.